python/paddle/fluid/dygraph/dygraph_to_static/program_translator.py

```diff
@@ -136,100 +142,345 @@ def convert_to_static(function):
         return static_func
 
 
-class FunctionSpec(object):
-    def __init__(self, func, args, kwargs):
-        self._dyfunc = func
-        self._args = args
-        self._kwargs = kwargs
+class CacheKey(object):
```

Comment:
If your cache logic is growing, we had better to split it to another python file in the long future (may not be this PR)

Reply:
Good idea. I will pull a new request to split it.